src/netaddress.cpp

```diff
@@ -515,6 +517,29 @@ bool CNetAddr::IsAddrV1Compatible() const
     assert(false);
 }
 
+bool CNetAddr::LimitOutboundByNetgroup() const
+{
+    const auto onlynets = gArgs.GetArgs("-onlynet");
+    if (onlynets.size() != 1) {
+        return true;
+    }
+    switch (m_net) {
```

Comment:
we would reach here when size of `onlynets` is 1. it's possible that `m_net` doesn't belong to the network we mention in `onlynets` but that would be handled in [reachability check](https://github.com/bitcoin/bitcoin/blob/47184cfa2f62290c00f960ddf74a7c1b5d3578b0/src/net.cpp#L1855) in `ThreadOpenConnections()` and we wouldn't use that address anyways.